### ISSUE TITLE
SmoothL1Loss and HuberLoss threshold on the signed difference, so every difference below -beta is wrong (and their backward pass raises)

### ISSUE THREAD

**Issue by truong-v** [BobMcDear/attorch#13]
## Summary

`SmoothL1Loss` and `HuberLoss` switch from their quadratic piece to their linear piece at a threshold on the **absolute** difference `|input - target|`. The kernels compare the **signed** difference against the threshold instead, so every difference at or below `-beta` (`-delta`) takes the quadratic branch where PyTorch takes the linear one — on ordinary data roughly a quarter of the elements, in both the forward and the backward pass. `L1Loss` and `MSELoss` have no threshold and are unaffected.

Separately, the backward pass of these two losses cannot run at all: they pass `beta`/`delta` as a fifth argument to `PLossAutoGrad.apply`, but `PLossAutoGrad.backward` returns four gradients, so autograd raises `returned an incorrect number of gradients (expected 5, got 4)` — the same slip that `1a3a517` fixed for layer norm. `tests/test_p_loss_layers.py` does catch this (it currently reports **596 failed, 796 passed**, 522 of those failures these two losses), but it never reaches its gradient assertions: it fails one line earlier, on the forward comparison.

## Reproduction

```python
import torch
from torch import nn

import attorch

inp = torch.tensor([-3.0, -1.5, -0.5, 0.0, 0.5, 1.5, 3.0], device='cuda')
tgt = torch.zeros_like(inp)
row = lambda t: '  '.join(f'{v:7.3f}' for v in t.tolist())

for cls in ('SmoothL1Loss', 'HuberLoss'):
    attorch_loss = getattr(attorch, cls)(reduction='none')
    print(cls)
    print('  input - target', row(inp - tgt))
    print('  torch         ', row(getattr(nn, cls)(reduction='none')(inp, tgt)))
    print('  attorch       ', row(attorch_loss(inp, tgt)))

    a_in = inp.clone().requires_grad_(True)
    out = attorch_loss(a_in, tgt)
    try:
        out.backward(torch.ones_like(out))
        print('  backward       ok')
    except RuntimeError as error:
        print('  backward      ', error)
```

```
SmoothL1Loss
  input - target  -3.000   -1.500   -0.500    0.000    0.500    1.500    3.000
  torch            2.500    1.000    0.125    0.000    0.125    1.000    2.500
  attorch          4.500    1.125    0.125    0.000    0.125    1.000    2.500
  backward       function PLossAutoGradBackward returned an incorrect number of gradients (expected 5, got 4)
HuberLoss
  input - target  -3.000   -1.500   -0.500    0.000    0.500    1.500    3.000
  torch            2.500    1.000    0.125    0.000    0.125    1.000    2.500
  attorch          4.500    1.125    0.125    0.000    0.125    1.000    2.500
  backward       function PLossAutoGradBackward returned an incorrect number of gradients (expected 5, got 4)
```

A control where every difference is non-negative — so the defect cannot fire — matches PyTorch exactly (`0.0000`).

## Environment

- attorch at `88ed34d`
- Reproduced on NVIDIA L40S (sm_89) and H100 NVL (sm_90)
- torch 2.4.0+cu121, triton 3.0.0, Python 3.10, driver CUDA 12.4
